formbuilder/components/XLSDownloader.js

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import json2xls from "json2xls";
+import btoa from "btoa";
+
+export default function XLSDownloader(props) {
+  const filename = props.schema.title + ".xls";
+  const fields = props.fields;
+  const fieldNames = props.fields.map((key) => {
+    return props.schema.properties[key].title;
+  });
+
+  // Pre-processing of data
+  // json2xls takes json in the following format
+  // json_data = [
+  //   {
+  //     col1: row1value1,
+  //     col2: row1value2
+  //   },
+  //   {
+  //     col1: row2value1,
+  //     col2: row2value2
+  //   }
+  // ]
+  // After the pre-processing is done, we will have data in
+  // above mentioned format in `formatted_data` which can be
+  // then easily fed to `json2xls`.
+  var formatted_data = [];
+  var rows = props.records;
```

Comment:
I think we should be able to iterate on the records in another way. Since it's a list of object, we can process them with a `map` function instead.